src/extension.ts

```diff
@@ -80,17 +82,55 @@ let decorationType: TextEditorDecorationType = window.createTextEditorDecoration
 
 const config = workspace.getConfiguration("metals");
 
+interface Activation {
+  shouldCreateNewProject: boolean;
+  status: Disposable;
+}
+
 export async function activate(context: ExtensionContext) {
   detectLaunchConfigurationChanges();
   checkServerVersion();
+  let activation: Activation = {
+    shouldCreateNewProject: false,
+    status: window.setStatusBarMessage("$(gear~spin) Activating Metals"),
+  };
+
+  context.subscriptions.push(
+    commands.registerCommand("metals.new-scala-project-activate", () => {
+      activation.shouldCreateNewProject = true;
+    })
+  );
 
   getJavaHome(workspace.getConfiguration("metals").get("javaHome"))
-    .then((javaHome) => fetchAndLaunchMetals(context, javaHome))
+    .then((javaHome) => fetchAndLaunchMetals(context, javaHome, activation))
     .catch((err) => {
       outputChannel.appendLine(err);
       showMissingJavaMessage();
+      activation.status.dispose();
     });
   commands.executeCommand("setContext", "metals:enabled", true);
+  setHasBuildTool();
+}
+
+function setHasBuildTool(): void {
+  let allPromises = [
+    "build.sbt",
+    " build.sc",
```

Comment:
Is this space intentional?
```suggestion
    "build.sc",
```

Reply:
No! Thanks a lot, no idea how it got there.